backend/services/office_hours/ticket.py

```diff
@@ -1,689 +1,280 @@
+        if not ticket_entity:
+            raise ResourceNotFoundException(f"Ticket not found with ID: {ticket_id}")
 
-        # Fetch Office Hours Section - Needed To Determine if User Membership
-        oh_section_entity = self._get_office_hours_section_by_oh_event_id(
-            ticket_entity.oh_event_id
-        )
+        if ticket_entity.state != TicketState.QUEUED:
+            raise CoursePermissionException(
+                "Cannot call a ticket that is not in the queue."
+            )
 
-        # Fetch Current User Section Membership
-        current_user_section_member_entity = self._check_user_section_membership(
-            subject.id, oh_section_entity.id
+        # Create query off of the member query for just the members matching
+        # with the current user (used to determine permissions)
+        user_member_query = (
+            select(SectionMemberEntity)
+            .where(SectionMemberEntity.user_id == user.id)
+            .join(SectionEntity)
+            .join(CourseSiteEntity)
+            .join(OfficeHoursEntity)
+            .where(OfficeHoursEntity.id == ticket_entity.office_hours_id)
         )
 
-        # If current user is student, check if they are one of the creators of ticket
-        # Raise Exception if Not Creator!
-        if current_user_section_member_entity.member_role == RosterRole.STUDENT:
-            ticket_creators = ticket_entity.to_details_model().creators
-            if current_user_section_member_entity.id not in [
-                creator.id for creator in ticket_creators
-            ]:
-                raise PermissionError(
-                    f"User Doesn't Have Permission to Get Ticket id={ticket_entity.id}"
-                )
-
-        # Passed Permissions - Good to Return Ticket Information
-        return ticket_entity.to_model()
-
-    def get_ticket_details_by_id(
-        self, subject: User, oh_ticket_id: int
-    ) -> OfficeHoursTicketDetails:
-        """
-        Retrieves an office hours ticket details from the table by its id.
+        user_member = self._session.scalars(user_member_query).unique().one_or_none()
 
-        Args:
-            subject (User): A valid User model representing the currently logged-in user.
-            oh_ticket_id (int): ID of the ticket to query by.
-
-        Returns:
-            OfficeHoursTicketDetails: `OfficeHoursTicketDetails` with the given id
-        """
-        # Fetch Ticket By ID
-        ticket_entity = self._session.get(OfficeHoursTicketEntity, oh_ticket_id)
-
-        if ticket_entity is None:
-            raise ResourceNotFoundException(
-                f"Office Hours Ticket with id={oh_ticket_id} not found."
+        # If the user is not a member of the looked up course, throw an error
+        if not user_member or user_member.member_role == RosterRole.STUDENT:
+            raise CoursePermissionException(
+                "Not allowed to call if a ticket if you are not a UTA, GTA, or instructor for."
```

Comment:
In the current state of the queue, where it's updated based on polling, we'll want to be very careful about how this exception is handled client side: it's more common than you'd think, esp in a course like 110 with 5+ TAs on staff, that two people will call the same ticket within a 10s refresh window. When this happens, the person who called second should have some kind of reasonable UX in not getting an error but a meaningful message that the ticket was already called.